infra/feast-operator/api/v1alpha1/featurestore_types.go

```diff
@@ -71,16 +71,60 @@ type FeatureStoreServices struct {
 // OfflineStore configures the deployed offline store service
 type OfflineStore struct {
 	ServiceConfigs `json:",inline"`
+	// +optional
+	Persistence *OfflineStorePersistence `json:"persistence,omitempty"`
+}
+
+// OfflineStorePersistence configures the persistence settings for the offline store service
+type OfflineStorePersistence struct {
+	// +optional
+	FilePersistence *OfflineStoreFilePersistence `json:"file,omitempty"`
+}
+
+// OfflineStorePersistence configures the file-based persistence for the offline store service
+type OfflineStoreFilePersistence struct {
+	// +optional
+	// +default:value=dask
+	// +kubebuilder:validation:Enum=dask;duckdb
+	Type string `json:"type,omitempty"`
 }
```

Comment:
if possible, would be cleaner to assign `OfflineConfigType` directly in the api. then the parse function and checks go away. this would require moving several constants and a var out of services types... to the api types file.
```suggestion
// OfflineStorePersistence configures the file-based persistence for the offline store service
type OfflineStoreFilePersistence struct {
	// +kubebuilder:validation:Enum=dask;duckdb
	Type OfflineConfigType `json:"type,omitempty"`
}

const (
	OfflineRemoteConfigType OfflineConfigType = "remote"
	OfflineDaskConfigType   OfflineConfigType = "dask"
	OfflineDuckDbConfigType OfflineConfigType = "duckdb"
)

// OfflineConfigType provider name or a class name that implements Offline Store
type OfflineConfigType string
```

Reply:
anyway, its worth trying. hopefully it negates the need for parsing function.

Reply:
That check was wrong, we should not check the value from CRD while converting to the one for RepoConfig, they are different things for now.
I added instead a validation function in the API for the supported file types of offline store, and used it in the ApplyDefaults function. Please TAL at the next commit